src/plugins/plugin.legend.js

```diff
@@ -237,26 +237,53 @@ var Legend = Element.extend({
 			minSize.height = me.maxHeight; // fill all the height
 		}
 
+		var getMaxLineWidth = function(boxWidth, textLines) {
+			return textLines.map(function(textLine) {
+				return boxWidth + (fontSize / 2) + ctx.measureText(textLine).width;
+			}).reduce(function(acc, v) {
+				return v > acc ? v : acc;
+			}, 0);
+		};
+
 		// Increase sizes here
 		if (display) {
 			ctx.font = labelFont.string;
 
 			if (isHorizontal) {
+
 				// Labels
 
 				// Width of each line of legend boxes. Labels wrap onto multiple lines when there are too many to fit on one
 				var lineWidths = me.lineWidths = [0];
+				var maxHeight = 0;
 				var totalHeight = 0;
 
 				ctx.textAlign = 'left';
 				ctx.textBaseline = 'top';
 
 				helpers.each(me.legendItems, function(legendItem, i) {
 					var boxWidth = getBoxWidth(labelOpts, fontSize);
-					var width = boxWidth + (fontSize / 2) + ctx.measureText(legendItem.text).width;
+					var width;
```

Comment:
You can put these all on one line:

    var width, height, grossHeight;

Reply:
Done